tensorrt_llm/_torch/pyexecutor/py_executor_creator.py

```diff
@@ -252,13 +252,31 @@ def create_py_executor(
         with mem_monitor.observe_creation_stage(
                 _ExecutorCreationStage.MODEL_ENGINE_DRAFT):
             draft_spec_config = copy.copy(spec_config)
-            draft_pytorch_backend_config = copy.copy(pytorch_backend_config)
-            if spec_config.load_format == "dummy":
-                draft_pytorch_backend_config.load_format = LoadFormat.DUMMY
             # The draft model won't have any draft tokens attached to
             # generation requests when we invoke it autoregressively
             draft_spec_config.max_draft_len = 0
 
+            use_chain_drafter = (
+                executor_config.guided_decoding_config is None
+                and not pytorch_backend_config.enable_mixed_sampler)
+
+            if use_chain_drafter:
+
+                # Make sure we allocate enough KV cache space.
+                spec_config._num_extra_kv_tokens = spec_config.max_draft_len - 1
```

Comment:
Note: this was an attempt to fix a bug where not enough KV cache was being allocated during CUDA graph warmup. It's not the right way to do this though; will revert it tomorrow.